src/backend/marsha/core/management/commands/test_mediapackage_harvest.py

```diff
@@ -0,0 +1,133 @@
+"""Test recording slicing management command."""
+import datetime
+from pprint import pprint
+import time
+
+from django.conf import settings
+from django.core.management.base import BaseCommand, CommandError
+
+from marsha.core.models import Video
+
+
+class Command(BaseCommand):
+    """Creates recording slices for a live and waits for the videos to be harvested."""
+
+    help = (
+        "Creates recording slices for a live and waits for the videos to be harvested."
+    )
+
+    def wait_seconds(self, seconds):
+        print("")
+        for seconds_left in range(seconds, 0, -1):
+            self.stdout.write(f"Waiting {seconds_left} seconds...", ending="\r")
+            time.sleep(1)
+
+    def add_arguments(self, parser):
+        """Add arguments to the command."""
+        parser.add_argument(
+            "-s", "--slices", type=int, default=3, help="Number of slices"
+        )
+        parser.add_argument(
+            "-l", "--length", type=int, default=30, help="Length of slices in seconds"
+        )
+        parser.add_argument(
+            "-w",
+            "--wait",
+            type=int,
+            default=20,
+            help="Number of seconds between slices",
+        )
+        parser.add_argument(
+            "-f", "--force", action="store_true", help="Force slices re-creation"
+        )
+
+    def handle(self, *args, **options):
+        """Execute management command."""
+
+        if not settings.DEBUG:
+            raise CommandError(
+                "This command can only be executed when settings.DEBUG is True."
+            )
+        slices = options["slices"]
+        length = options["length"]
+        wait = options["wait"]
+        force = options["force"]
+
+        self.stdout.write(
+            f"\nCreating {slices} slices of {length} seconds each "
+            f"with a wait of {wait} seconds between slices."
+        )
+
+        video = Video.objects.order_by("created_on").last()
+        self.stdout.write(f"\nTesting recording slices for {video.id} {video.title}")
+
+        if force:
+            self.stdout.write("\nForcing slices re-creation")
+            video.recording_slices = None
+            video.save()
+
+        self.wait_video_state(video, "live_state", "running")
+
+        self.wait_seconds(5)
+
+        self.stdout.write("\nCreating recording slices...")
+        # breakpoint()
+
+        # ask user if he wants to start recording with default to no
+        start_recording = input("\nStart recording? (y/N)")
```

Comment:
This question is perturbing when I use the command without modifying any default arguments. I expected to have 3 recording slices and I thought the command was asking me if I want to process the 3 expected recording slices. But in fact it manually start and stop a recording slice, then the 3 expected recording_slice will never be created because you are checking line 85 if there are already recording slice existing...
So, does it matter to keep this and the check you are doing line 85 ?

Reply:
I added this to test the harvesting behavior while pausing a streaming, but it surely needs a rework. :+1: